pykern/pkunit.py

```diff
@@ -510,20 +514,44 @@ def _actual_xlsx_to_csv(self, actual_xlsx, sheet):
             line_terminator="\r\n",
         )
 
+    def _gen_ignore_lines(self):
+        r = []
+        if self._ignore_lines is None:
+            return r
+        for l in (l for l in [l for l in self._ignore_lines if l]):
```

Comment:
There are too many `for` loops. This is a better encapsulation, which both shortens the code and separates out a cohesive chunk:

```py
def _cmd():
    if self.is_bytes:
        r = ["cmp"]
    else:
        r = ["diff"]
        for l in self._ignore_lines or ():
            r.extend(("-I", l))
    return r + [str(self._expect_path), str(self._actual_path)]
```

Move `_cmd` inside `_compare`. You don't need to pass self.

`_gen_ignore_lines` is non-descriptive. 

`ignore_lines` could be [], (), None, False, etc. We don't care, it's falsey, and do nothing. 

Use tuples unless you expect the object is going to be modified. 

The elements of ignore_lines must be strings. If someone wants to pass "" as an element, pass that as a value to -I.